### PR TITLE
search project

### PR DESCRIPTION
**Pull request by BelehovEgor**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/624

### CODE REVIEW

File: src/SearchService.Bussines/Commands/Search/SearchCommand.cs [hunk 1]
```diff
@@ -1,14 +1,87 @@
...
+                    return projects;
+                }
+
+                _logger.LogWarning(errorMessage);
+
+                errors.Add(errorMessage);
+            }
+            catch (Exception exc)
+            {
+                _logger.LogWarning(exc, errorMessage);
```
Comment (djino97) [inline src/SearchService.Bussines/Commands/Search/SearchCommand.cs]: Why logWarning instead logError?

File: test/SearchService.Bussines.UnitTests/SearchCommandTests.cs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                        It.IsAny<object>(), default, default))
+                .Returns(Task.FromResult(_brokerSearchResponseMock.Object));
+
+            var expectedResponse = new SearchResponse
+            {
+                Items = new List<SearchResultInfo>(),
+                Errors = new List<string> { "Cannot search projects. Please try again later." }
+            };
+
+            SerializerAssert.AreEqual(expectedResponse, _command.Execute("Project", new SearchFilter { IncludeProjects = true }));
```
Comment (djino97) [inline test/SearchService.Bussines.UnitTests/SearchCommandTests.cs:233]: need add Verify to tests

File: src/SearchService.Bussines/Commands/Search/SearchCommand.cs [hunk 3]
```diff
@@ -1,14 +1,87 @@
...
+                            {
+                                Type = SearchResultObjectType.Project,
+                                Id = p.Id,
+                                Value = p.Value
+                            }).ToList());
+
+                    return projects;
+                }
+
+                _logger.LogWarning(errorMessage);
```
Comment (djino97) [inline src/SearchService.Bussines/Commands/Search/SearchCommand.cs:53]: add response.MessageErrors to Logger

File: test/SearchService.Bussines.UnitTests/SearchCommandTests.cs (hunk 2 above)
Comment (BelehovEgor) [reply test/SearchService.Bussines.UnitTests/SearchCommandTests.cs:233]: fixed

File: src/SearchService.Bussines/Commands/Search/SearchCommand.cs (hunk 3 above)
Comment (BelehovEgor) [reply src/SearchService.Bussines/Commands/Search/SearchCommand.cs:53]: here should be warning

File: src/SearchService.Bussines/Commands/Search/SearchCommand.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/SearchService.Bussines/Commands/Search/SearchCommand.cs]: fixed

File: src/SearchService.Bussines/Commands/Search/SearchCommand.cs [hunk 4]
```diff
@@ -1,14 +1,87 @@
...
     {
-        public SearchResponse Execute(string text)
+        private IRequestClient<ISearchProjectsRequest> _rcProjects;
+        private ILogger<SearchCommand> _logger;
+
+        private List<SearchResultInfo> SearchProjects(string text, List<string> errors)
+        {
+            List<SearchResultInfo> projects = new();
+
+            string errorMessage = "Cannot search projects. Please try again later.";
```
Comment (djino97) [inline src/SearchService.Bussines/Commands/Search/SearchCommand.cs:32]: add text variable in message

File: src/SearchService.Bussines/Commands/Search/SearchCommand.cs [hunk 5]
```diff
@@ -1,14 +1,87 @@
...
-            throw new NotImplementedException();
+            _rcProjects = rcProjects;
+            _logger = logger;
+        }
+
+        public SearchResponse Execute(string text, SearchFilter filter)
+        {
+            List<string> errors = new();
+
+            List<SearchResultInfo> response = new();
```
Comment (djino97) [inline src/SearchService.Bussines/Commands/Search/SearchCommand.cs:163]: Add in SearchResponse model errors initialization

**Review by djino97 — APPROVED** [review]

File: src/SearchService.Bussines/Commands/Search/SearchCommand.cs (hunk 5 above)
Comment (BelehovEgor) [reply src/SearchService.Bussines/Commands/Search/SearchCommand.cs:163]: what?
what do you mean?

File: src/SearchService.Bussines/Commands/Search/SearchCommand.cs (hunk 4 above)
Comment (BelehovEgor) [reply src/SearchService.Bussines/Commands/Search/SearchCommand.cs:32]: what do you want to add here?

**Review by MariyaSin — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

**Review by TIove — APPROVED** [review]

**Pull request merged**